### PR TITLE
fixed double key exception

### PR DESCRIPTION
**Pull request by fr8tra1n**

### CODE REVIEW

**Comment by jmbeach** [comment]
Can you give more insight into what issue this change is fixing?

**Comment by fr8tra1n** [comment]
Sorry to keep you waiting.
We experienced a case where the foldercache already contained the item resulting in an exception. This change fixed it.

**Comment by jmbeach** [comment]
That's a lot of changes at once. I still might merge them. I'm going to test them locally

File: h-opc/Ua/UaClient.cs [hunk 1]
```diff
@@ -450,7 +453,17 @@ private Session InitializeSession(Uri url)
         else
           throw new OpcException(string.Format("Failed to validate certificate with error code {0}: {1}", eventArgs.Error.Code, eventArgs.Error.AdditionalInfo), eventArgs.Error.StatusCode);
       };
-
+      
+      /*
+      var traceConfiguration = new TraceConfiguration
+      {
+          OutputFilePath = @"c:\temp\opc-ua.log",
```
Comment (jmbeach) [inline h-opc/Ua/UaClient.cs]: I would not like to have commented code source controlled

**Comment by jmbeach** [comment]
@fr8tra1n, Did you see my message? If you delete the commented out code, I'll merge.

**Comment by fr8tra1n** [comment]
I have cleaned the code as requested.
While the extra options gives the opc client more control it does not completely resolve timeouts resulting in exceptions such as BadSubscriptionIdInvalid. We experienced these exceptions on heavy subscription use (80+). After troubleshooting and experimenting (such as async vs. sync) we were able to resolve the issues by creating a single subscription with 80+ items to monitor. All with sync calls and without any non-default options.
In addition we are now referencing https://github.com/OPCFoundation/UA-.NET/releases/tag/1.03.340 instead of the older binaries. 
Because of the changes and resulting api-breaks we have not updated our github fork.

**Comment by jmbeach** [comment]
Gotcha. So you're using a newer version of OPC Foundation's UA client with h-opc?

**Comment by jmbeach** [comment]
I'll roll out a tag soon with this change

**Pull request merged**